server/index.mjs

```diff
@@ -73,6 +67,63 @@ app.get('/api/song/:id/playlists', (req, res) => {
   });
 });
 
+//* ************************* NEW ENDPOINTS **************************** *//
+
+app.get('/api/song/:id/retrieve', (req, res) => {
+  connection.query(`select ${params.id} from songs`, (err, dbres) => {
+    if (err) {
+      res.json(err);
+    }
+    res.status(200).json(dbres);
+  });
+});
+
+app.put('/api/song/:id/update', (req, res) => {
+  //NOTE: for PUT/UPDATE, data validation should occur with an option to return an error BAD REQUEST (400)
+  //not worth doing right now
+
+  //the below might be an async wait, but I don't want to worry about that right now since it wont be used.
+  let song = req.data;
+
+  let updateString = '';
+  for (let key in song) {
+    updateString += `${key} = ${song[key]},`;
+  }
+
+  let sql = `UPDATE songs SET (${updateString}) WHERE id = ${params.id}`;
+
+  connection.query(sql, (err, res) => {
+    if (err) {
+      console.log(err);
+      res.status(500).end();
+    }
+    res.status(200).end();
+  });
+});
+
+app.post('/api/song/:id/create', (req, res) => {
+  let song = req.data;
+  let sql = `INSERT INTO songs(${Object.keys(song).slice(1).join(',')}) VALUES(${Object.values(song).slice(1).map(value => (!isBoolean(value) ? `"${value}"` : value ? '1' : '0')).join(',')})`;
+  connection.query(sql, (err, res) => {
+    if (err) {
+      console.log(err);
+      res.status(500).end();
+    }
+    res.status(201).end();
+  });
+});
+
+app.delete('/api/song/:id/delete', (req, res) => {
```

Comment:
The requests these functions are expecting should handle the routing without needs to be repeated in the url path.

Reply:
Sorry, I don't understand what you mean.  Can you please elaborate?